tailcfg/tailcfg.go

```diff
@@ -2048,7 +2049,59 @@ type SSHAction struct {
 
 	// Recorders defines the destinations of the SSH session recorders.
 	// The recording will be uploaded to http://addr:port/record.
-	Recorders []netip.AddrPort `json:"recorders"`
+	Recorders []netip.AddrPort `json:"recorders,omitempty"`
+
+	// OnRecorderFailure is the action to take if recording fails.
+	// If nil, the default action is to fail open.
+	OnRecordingFailure *SSHRecorderFailureAction `json:"onRecordingFailure,omitempty"`
+}
+
+// SSHRecorderFailureAction is the action to take if recording fails.
+type SSHRecorderFailureAction struct {
+	// RejectSession, if true, specifies that the session should
+	// be rejected if the recording fails to start.
+	RejectSession bool `json:",omitempty"`
+
+	// TerminateSession, if true, specifies that the session should
+	// be terminated if the recording fails after it has started.
+	TerminateSession bool `json:",omitempty"`
```

Comment:
I think these two fields could be more explicitly named, but I don't have better ideas that aren't super long in the few minutes of noodling i've been doing.

Reply:
eh, docs are clear no?